### PR TITLE
Add logging of encrypted buffer per ELG-252 using sprintf

### PR DESCRIPTION
**Pull request by gcleary**
Note that I applied the same simplification to logfmt.

### CODE REVIEW

**Comment by gcleary** [comment]
New sample dump:

Skyhook libEL DEBG: libel.c:sky_finalize_request()Request buffer of 224 bytes prepared
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000000: 0B 08 02 10 14 18 C0 01 20 02 28 01 0A 10 7C C2
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000010: 54 F8 1B E8 E7 8D 76 5A 2E 63 33 9F C9 9A 10 09
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000020: 51 E9 BE 9C FE 04 D9 68 92 AD 61 C0 E5 B9 DC FF
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000030: 12 25 65 2B 46 0B F7 10 34 1E 1D A7 6B 20 37 89
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000040: DA 48 76 26 45 42 47 6E F9 17 20 FC A6 40 5F D5
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000050: 49 23 63 03 79 44 A8 6A DC EF F7 1D 07 FD 29 45
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000060: 29 50 33 1E F9 EE E8 26 9A 48 BD 4A BA 85 8C 77
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000070: EB C5 95 C9 F1 08 18 87 8E 8A F9 D0 6E EB 59 A1
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000080: 39 4D 91 C4 89 73 CF 6C 6C A4 47 8C B6 D7 30 86
Skyhook libEL DEBG: libel.c:sky_finalize_request() 0000090: 05 D3 23 DD 85 D2 E0 CE CD A6 A4 D8 4C 73 61 47
Skyhook libEL DEBG: libel.c:sky_finalize_request() 00000A0: 8B 20 28 B7 4B 05 D5 D4 4A 09 1C 01 95 90 7C 03
Skyhook libEL DEBG: libel.c:sky_finalize_request() 00000B0: 4D 71 10 5D 08 D1 A6 92 BB E3 13 3A F1 29 A6 5B
Skyhook libEL DEBG: libel.c:sky_finalize_request() 00000C0: C5 A0 32 93 0C 24 BB ED 3C 67 35 15 13 55 4B EC
Skyhook libEL DEBG: libel.c:sky_finalize_request() 00000D0: 97 3F 3A 00 D9 6E 50 7C 5E B2 72 DA 87 2F 8B C8

File: libel/utilities.c [hunk 1]
```diff
@@ -226,6 +224,61 @@ int logfmt(
...
+    void *buffer, uint32_t bufsize, int index)
+{
+#if SKY_DEBUG
+    char buf[SKY_LOG_LENGTH];
+    uint8_t *b = (uint8_t *)buffer;
+    int n, N, pb;
+    if (level > ctx->min_level || function == NULL || buffer == NULL || bufsize <= 0)
+        return -1;
+    memset(buf, '\0', sizeof(buf));
+    n = snprintf(buf, sizeof(buf), "%.20s:%.20s() %07X:", sky_basename(file), function, index);
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Perhaps a comment above this like
```c
// Print log-line prefix ("<source file>:<function name> <buf offset>:")
```
would make clear at a glance what this line does.

File: libel/utilities.c [hunk 2]
```diff
@@ -226,6 +224,61 @@ int logfmt(
 }
 #endif
 
+/*! \brief dump maximum number of bytes of the given buffer in hex on one line
+ *
+ *  @param buf pointer to the buffer
+ *  @param bufsize size of the buffer in bytes
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Need descriptions for other parameters.

File: libel/utilities.c [hunk 3]
```diff
@@ -226,6 +224,61 @@ int logfmt(
...
+ *
+ *  @returns number of bytes dumpped, or negitive number on error
+ */
+int dump_hex16(const char *file, const char *function, Sky_ctx_t *ctx, Sky_log_level_t level,
+    void *buffer, uint32_t bufsize, int index)
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Would `buf_offset` be a better name than `index`?

File: libel/utilities.c [hunk 4]
```diff
@@ -226,6 +224,61 @@ int logfmt(
 }
 #endif
 
+/*! \brief dump maximum number of bytes of the given buffer in hex on one line
+ *
+ *  @param buf pointer to the buffer
+ *  @param bufsize size of the buffer in bytes
+ *
+ *  @returns number of bytes dumpped, or negitive number on error
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Typo.

File: libel/utilities.c [hunk 5]
```diff
@@ -226,6 +224,61 @@ int logfmt(
...
+#if SKY_DEBUG
+    char buf[SKY_LOG_LENGTH];
+    uint8_t *b = (uint8_t *)buffer;
+    int n, N, pb;
+    if (level > ctx->min_level || function == NULL || buffer == NULL || bufsize <= 0)
+        return -1;
+    memset(buf, '\0', sizeof(buf));
+    n = snprintf(buf, sizeof(buf), "%.20s:%.20s() %07X:", sky_basename(file), function, index);
+
+    N = n + (16 * 3 - 1); /* 16 bytes per line (' XX') */
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Need a comment above this line indicating what `N` represents. Looks like it's the number of characters to print?

File: libel/utilities.c [hunk 6]
```diff
@@ -226,6 +224,61 @@ int logfmt(
...
+    /* if width of log line (SKY_LOG_LENGTH) too short for pretty print */
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: "if width ... for pretty print" then _what_? I'm left hanging after reading this comment.

File: libel/utilities.c [hunk 7]
```diff
@@ -241,10 +247,12 @@ int dump_hex16(const char *file, const char *function, Sky_ctx_t *ctx, Sky_log_l
...
     memset(buf, '\0', sizeof(buf));
-    n = snprintf(buf, sizeof(buf), "%.20s:%.20s() %07X:", sky_basename(file), function, index);
+    // Print log-line prefix ("<source file>:<function name> <buf offset>:")
+    n = snprintf(buf, sizeof(buf), "%.20s:%.20s() %07X:", sky_basename(file), function, buf_offset);
 
-    N = n + (16 * 3 - 1); /* 16 bytes per line (' XX') */
-    /* if width of log line (SKY_LOG_LENGTH) too short for pretty print */
+    // Calculate number of characters required to print 16 bytes
+    N = n + (16 * 3); /* 16 bytes per line, 3 bytes per byte (' XX') */
+    // if width of log line (SKY_LOG_LENGTH) too short 16 bytes, just pring those that fit
```
Comment (tboinske-skyhook) [inline libel/utilities.c]: Typo here ("pring")

**Comment by gcleary** [comment]
Thanks. Will merge after minor fix.

**Pull request merged**